script.js

```diff
@@ -1,12 +1,92 @@
 //You can edit ALL of the code here
+//Variables
+const rootElem = document.getElementById("root");
+const allEpisodes = getAllEpisodes();
+const selectEpisode = document.getElementById("allEpisodes");
+let searchBox = document.getElementById("search");
 function setup() {
-  const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  listEpisodes(allEpisodes);
+  // Item search function by using an event Listener
+  searchBox.addEventListener("input", () => {
+    let resultNumber = document.getElementById("resultNumber");
+    let results = allEpisodes.filter(containsSearchTerm);
+    makePageForEpisodes(results);
+    resultNumber.innerText = `${results.length} / ${allEpisodes.length}`;
+    if (searchBox.value.length == 0) {
+      resultNumber.innerText = "";
+    }
+  });
+  selectEpisode.addEventListener("change", () => {
+    if (selectEpisode.value === "novalue") {
+      makePageForEpisodes(allEpisodes);
+    } else {
+      let result = allEpisodes.find((episode) => {
+        return episode.id == selectEpisode.value;
+      });
+      makePageForEpisodes(result);
+    }
+  });
```

Comment:
The select episode is selecting but not showing the selected option. I'm having the same issue so I don't have any recommendations at the moment.